core/src/main/java/org/hibernate/ogm/boot/model/naming/impl/OgmImplicitNamingStrategy.java

```diff
@@ -23,19 +21,13 @@
 	/**
 	 * A pattern common to all property names used in element collections.
 	 */
-	private static final Pattern ELEMENT_COLLECTION_NAME_PATTERN = Pattern.compile( "collection&&element\\." );
+	private static final String ELEMENT_COLLECTION_NAME_PATTERN = "collection&&element\\.";
 
 	@Override
 	protected String transformAttributePath(AttributePath attributePath) {
 		// for element collections just use the simple name
-		String[] parts = ELEMENT_COLLECTION_NAME_PATTERN.split( attributePath.getFullPath() );
-
-		// for element collections just use the simple name
-		if ( parts.length == 2 ) {
-			return parts[1];
-		}
-		else {
-			return attributePath.getFullPath();
-		}
+		String fullPath = attributePath.getFullPath();
+		String attribute = fullPath.replaceAll( ELEMENT_COLLECTION_NAME_PATTERN, "" );
+		return attribute;
```

Comment:
BTW, if you end up doing something like that, the idea is to not mess with strings anymore. You have to navigate in the `AttributePath` hierarchy and use the `isCollectionElement` method. Steve plans to change the `collection&&element` part in a near future.